src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -536,6 +547,70 @@ AppendShardIdToConstraintName(AlterTableCmd *command, uint64 shardId)
 }
 
 
+static bool
+ExtendRowReferencesWalker(Node *node, RowRefWalkerState *state)
+{
+	if (node == NULL)
+	{
+		return false;
+	}
+
+	if (IsA(node, IndexElem))
```

Comment:
After checking out `9.5.0`, I see what you mean about `raw_expression_tree_walker` not understanding `IndexElem`. As written, this'll result in the `expr` being processed twice, though I suppose that's OK?